bundles/org.openhab.binding.serial/src/main/java/org/openhab/binding/serial/internal/handler/SerialDeviceHandler.java

```diff
@@ -50,29 +54,27 @@ public class SerialDeviceHandler extends BaseThingHandler {
 
     private final Map<ChannelUID, DeviceChannel> channels = new HashMap<>();
 
+    private final Map<ChannelUID, ScheduledFuture<?>> futures = new HashMap<>();
+
     public SerialDeviceHandler(final Thing thing) {
         super(thing);
     }
 
     @Override
     public void handleCommand(final ChannelUID channelUID, final Command command) {
         if (command instanceof RefreshType) {
-            final String lastValue = this.lastValue;
-
-            if (lastValue != null) {
-                final DeviceChannel channel = channels.get(channelUID);
-                if (channel != null) {
-                    refresh(channelUID, channel, lastValue);
-                }
+            final DeviceChannel channel = channels.get(channelUID);
+            if (channel != null) {
+                refresh(channelUID, channel);
             }
         } else {
             final DeviceChannel channel = channels.get(channelUID);
             if (channel != null) {
                 final Bridge bridge = getBridge();
                 if (bridge != null) {
```

Comment:
This would make assignment and cast at line 75 and the if check in line 76 obsolete.
```suggestion
                if (bridge != null && bridge.getHandler() instanceof CommonBridgeHandler handler) {
```

Reply:
Thanks for this hint, I just have learned a new Java syntax I didn't knew before :-)